### ISSUE TITLE
column lineages from boolean expression

### ISSUE THREAD

**Issue by joergraschke1205** [reata/sqllineage#236]
I have problems with some queries which contain boolean expressions (exasol-db). I only use the LineageRunner(sql_statement).get_column_lineage()-function without the gui.

The column lineages ignore the second boolean field (IS_CASE_TWO).
```
CREATE VIEW VIEW_A AS
SELECT
    TABLE_ID,
    NUMBER_01 + NUMBER_02 > 0 AS IS_CASE_ONE,
    (NUMBER_01 + NUMBER_02 > 0 AND NUMBER_03 + NUMBER_04 <= 0) AS IS_CASE_TWO
FROM
    TABLE_A
```

It is possible the following query parse with version 1.3.3. This ignore the timestamp fields in boolean expressions.
```
CREATE VIEW VIEW_B AS
SELECT
    TABLE_ID,
    (
        FIRST_TIMESTAMP IS NOT NULL
        AND FIRST_TIMESTAMP >= TO_TIMESTAMP(ADD_DAYS(CURRENT_DATE, -100))
        AND OTHER_TIMESTAMP > TO_TIMESTAMP(ADD_YEARS(CURRENT_DATE, -7))
    ) AS IS_QUALIFIED
FROM
    TABLE_B
```
I get in version 1.3.4 to many column lineages, like TABLE_B.IS -> IS_QUALIFIED or TABLE_B.NOT NULL -> IS_QUALIFIED.

**Comment by reata** [comment]
Sorry for the late reply. Due to the covid-19 situation being really bad in my city, didn't get the chance to dig deep into this.

I think this is probably related to #233 and #213 . Edge cases concerning parenthesis are not well tested. And we do have some regression issues in v1.3.4. Let me see if I can fix this and release a v1.3.5 this weekend.

**Comment by reata** [comment]
turns out this is a separate issue which has nothing to do with parenthesis. The impacted SQL are just boolean expressions as titled here. A new PR is created.
